### PR TITLE
feat: add optional PII/credential redaction to audit log entries

### PR DESCRIPTION
**Pull request by Ratnaditya-J**
## Summary

`buildAuditEntry()` in `mcp-proxy-server.ts` passes `argsForTransport` and `result.content` directly to the JSONL writer. Arguments without a resource role annotation (free-text content like file bodies, email text) hit the fallback at `path-utils.ts:181` and pass through with zero transformation.

This means if the agent reads a file containing an SSN, credit card number, or API key and passes that content as an argument to the next tool call, it lands in the audit log verbatim.

This PR adds a redaction layer that masks PII/credential patterns before writing audit entries.

New file: `src/trusted-process/audit-redactor.ts`
- Credit card detection with Luhn checksum validation (keeps first/last 4 digits)
- US Social Security Number detection with area/group/serial validation (keeps last 4)
- API key detection: OpenAI (`sk-`), GitHub PAT (`ghp_`), Slack (`xoxb-`/`xoxp-`), AWS (`AKIA`)
- Deep object walker that redacts string values at any nesting depth

Modified:
- `src/trusted-process/audit-log.ts` - `AuditLog` constructor accepts optional `{ redact: boolean }`. When enabled, redacts `arguments` and `result.content`/`result.error` before writing. Metadata fields (timestamps, tool names, policy decisions) are never modified.
- `src/config/user-config.ts` - New `auditRedaction` config section following the existing `autoApprove` pattern. Zod schema, defaults (`enabled: true`), resolved interface.
- `src/config/config-command.ts` - `auditRedaction` added to `computeDiff` nested sections for config editor visibility.
- `src/sandbox/index.ts` - Passes `AUDIT_REDACTION=true` env var to proxy subprocess when enabled.
- `src/trusted-process/mcp-proxy-server.ts` - Reads `AUDIT_REDACTION` env var, passes to `AuditLog` constructor.
- `src/trusted-process/index.ts` - Reads `config.userConfig.auditRedaction.enabled`, passes to `AuditLog`.

New file: `test/audit-redactor.test.ts`
- 28 tests covering all detection patterns, edge cases, deep object redaction, and AuditLog integration

### Design decisions

- **Enabled by default.** Redaction is on unless the user explicitly disables it. PII protection should be the safe default.
- **Redaction at the writer level.** The `AuditLog.log()` method applies redaction just before `JSON.stringify`, so all code paths that log audit entries get redaction automatically.
- **No new dependencies.** Pure regex + Luhn, same approach as the existing `redactCredentials()` in session log handling.

### Remaining items

- Missing test for `result.error` redaction path.

## Config

```json
{
  "auditRedaction": {
    "enabled": true
  }
}
```

## Test plan

- [x] 28 new tests pass (`npx vitest run test/audit-redactor.test.ts`)
- [x] Existing proxy server tests, user config tests still pass
- [x] TypeScript type-checks cleanly (`tsc --noEmit`)
- [x] 1243 total tests pass

Relates to #16

### CODE REVIEW

File: src/config/user-config.ts [hunk 1]
```diff
@@ -34,6 +34,9 @@ export const USER_CONFIG_DEFAULTS = {
     enabled: false,
     modelId: 'anthropic:claude-haiku-4-5',
   },
+  auditRedaction: {
+    enabled: false,
```
Comment (provos) [inline src/config/user-config.ts]: I think the safe default is to enable this by default. There aren't really many existing users to worry about.

File: src/trusted-process/mcp-proxy-server.ts [hunk 2]
```diff
@@ -338,6 +340,7 @@ export function parseProxyEnvConfig(): ProxyEnvConfig {
   delete process.env.SERVER_CREDENTIALS;
 
   const sandboxPolicy = (process.env.SANDBOX_POLICY ?? 'warn') as SandboxAvailabilityPolicy;
+  const auditRedaction = process.env.AUDIT_REDACTION === 'true';
```
Comment (provos) [inline src/trusted-process/mcp-proxy-server.ts:343]: should default to true
